ci/scripts/utils/githubpr.py

```diff
@@ -0,0 +1,125 @@
+
+
+class GitHubPR(Github):
+    """
+    GitHubPR is an inherited class from GitHub in pyGitHub for interacting with GitHub pull requests.
+
+    Attributes
+    ----------
+    repo : github.Repository.Repository
+        The GitHub repository to interact with.
+    pulls : github.PaginatedList.PaginatedList of github.PullRequest.PullRequest
+        The list of open pull requests in the repository, sorted by last updated.
+    user : github.AuthenticatedUser.AuthenticatedUser
+        The authenticated user.
+    InputFileContent : github.InputFileContent.InputFileContent
+        The class used to create file content for gists.
+
+    Methods
+    -------
+    __init__(self, repo_url=None, TOKEN=None)
+        Initialize a new GitHubPR instance.
+    get_repo_url(self, repo_url=None)
+        Set the repository for the GitHubPR instance
+        using an URL directly or from 'REPO_URL' environment variable.
+    get_pr_list(self)
+        Get the numerical list of all pull requests.
+    get_ci_pr_list(self, state='Ready', host=None)
+        Get the numerical list of all pull requests with a specific state from labels.
+        for example if a PR has a label 'CI-Ready-Hera' of the form CI-[state]-[host]
+        its corresponding PR number will be included in the list.
+    """
+
+    def __init__(self, repo_url=None, TOKEN=None):
+        """
+        __init__ Initialize a new GitHubPR instance.
+
+        This method authenticates with the GitHub API using the 'gh' CLI tool
+        when the TOKEN is not provided. The repository comes from from the 'REPO_URL'
+        environment variable when repo_url is not provided.
+        """
+        if TOKEN is None:
+            gh_cli = which('gh')
+            gh_cli.add_default_arg(['auth', 'status', '--show-token'])
+            TOKEN = gh_cli(output=str, error=str).split('\n')[3].split(': ')[1]
+        super().__init__(TOKEN)
+
+        self.repo = self.get_repo_url(repo_url)
+        self.pulls = self.repo.get_pulls(state='open', sort='updated', direction='desc')
+        self.user = self.get_user()
+
+        self.InputFileContent = InputFileContent
+
+    def get_repo_url(self, repo_url=None):
+        """
+        set_repo Set the repository for the GitHubPR instance.
+
+        Parameters
+        ----------
+        repo_url : Repository URL
+            The GitHub repository.
+        """
+        if repo_url is None:
+            repo_url = os.environ.get("REPO_URL")
+        match = re.search(r"github\.com/(.+)", repo_url)
+        repo_identifier = match.group(1)[:-4]
+        return self.get_repo(repo_identifier)
+
+    def get_pr_list(self):
+        """
+        get_pr_list Get the numerical list of all pull requests.
+
+        Returns
+        -------
+        list
+            A list of all pull request numbers.
+        """
+        pr_list = []
+        for pull in self.pulls:
+            pr_list.append(pull.number)
+        return pr_list
```

Comment:
This can be one line using list notation (does it even need to be a method?)
```suggestion
        return [ pull.number for pull in self.pulls ]
```